content/api/commands/origin.md

```diff
@@ -0,0 +1,512 @@
+  // ...and receive them at the other end here!
+  ({ username, password }) => {
+    cy.visit('/login')
+    cy.get('input#username').type(username)
+    cy.get('input#password').type(password)
+    cy.contains('button', 'Login').click()
+  }
+)
+```
+
+Note: You can just replace `sentArgs` with `args` if you want, the naming in
+this example is purely for clarity.
+
+### Yielding a value
+
+Values returned or yielded from the callback function **must** be serializable
+or they will not be returned to the primary origin. For example, the following
+will not work:
+
+**<Icon name="exclamation-triangle" color="red"></Icon> Incorrect Usage**
+
+```js
+cy.origin('https://www.acme.com', () => {
+  cy.visit('/')
+  cy.get('h1') // Yields an element, which can't be serialized...
+}).contains('ACME CORP') // ...so this will fail
+```
+
+Instead, you must explicitly yield a serializable value:
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```js
+cy.origin('https://www.acme.com', () => {
+  cy.visit('/')
+  cy.get('h1').invoke('textContent') // Yields a string...
+}).should('equal', 'ACME CORP') // 👍
+```
+
+### Navigating to secondary origin with cy.visit
+
+When navigating to a secondary origin using `cy.visit()`, it is essential to
+trigger the navigation **after** entering the origin callback, otherwise a
+cross-domain error will be thrown.
+
+```js
+// Do things in primary domain...
+
+cy.origin('www.acme.com', () => {
+  // Visit https://www.acme.com/history/founder
+  cy.visit('/history/founder')
+  cy.get('h1').contains('About our Founder, Marvin Acme')
+})
+```
+
+Here the `baseUrl` inside the callback is set to `www.acme.com` and the protocol
+defaults to `https`. When `cy.visit()` is called with the path
+`/history/founder`, the three are concatenated to make
+`https://www.acme.com/history/founder`.
+
+### Navigating to secondary origin with UI
+
+When navigating to a secondary origin by clicking a link or button in the
+primary origin, it is essential to trigger the navigation _before_ entering the
+origin callback, otherwise a cross-domain error will be thrown.
+
+```js
+// Button in primary origin goes to https://www.acme.com
+cy.contains('button', 'Go').click()
+
+cy.origin('www.acme.com', () => {
+  // No cy.visit is needed as the button brought us here
+  cy.get('h1').contains('ACME CORP')
+})
+```
+
+### Navigating to multiple secondary origins in succession
+
+Callbacks may **not** themselves contain `cy.origin()` calls, so when visiting
+multiple origins, do so at the top level of the test.
+
+```js
+cy.origin('foo.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'foo.com')
+})
+
+cy.origin('bar.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'bar.com')
+})
+
+cy.origin('baz.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'baz.com')
+})
+```
+
+<Alert type="info">
+
+A future version of Cypress will allow the use of nested `cy.origin()` calls.
+
+</Alert>
```

Comment:
Should we remove this notice? When we do the work, we can announce it but this will cause people to ask for when this will be supported.